ambry-server/src/integration-test/java/com.github.ambry.server/RouterServerTestFramework.java

```diff
@@ -591,7 +662,14 @@ public void onCompletion(T result, Exception exception) {
         return;
       }
       action(result);
-      continueChain(opChain);
+      try {
+        continueChain(opChain);
+      } catch (IOException e) {
+        if (expectError) {
```

Comment:
! (not)

Reply:
This way to throw `IOException` does't work, it just terminated the test earlier. Test passed if I throw an `IOException`. 
It's `TestFuture.check()` do the check, so I tried another way which throw `IOException` in `check()`

Reply:
The new way can throw sth like this;

```
java.io.IOException: hahaha

	at com.github.ambry.server.RouterServerTestFramework.startGetBlobAuthorizationFailTest(RouterServerTestFramework.java:367)
	at com.github.ambry.server.RouterServerTestFramework.continueChain(RouterServerTestFramework.java:504)
	at com.github.ambry.server.RouterServerTestFramework.access$300(RouterServerTestFramework.java:65)
	at com.github.ambry.server.RouterServerTestFramework$TestCallback.onCompletion(RouterServerTestFramework.java:695)
	at com.github.ambry.router.NonBlockingRouter$1.onCompletion(NonBlockingRouter.java:153)
	at com.github.ambry.router.NonBlockingRouter$1.onCompletion(NonBlockingRouter.java:147)
	at com.github.ambry.router.NonBlockingRouter.completeOperation(NonBlockingRouter.java:465)
	at com.github.ambry.router.NonBlockingRouter.completeOperation(NonBlockingRouter.java:442)
	at com.github.ambry.router.GetBlobInfoOperation.checkAndMaybeComplete(GetBlobInfoOperation.java:437)
	at com.github.ambry.router.GetBlobInfoOperation.handleResponse(GetBlobInfoOperation.java:251)
	at com.github.ambry.router.GetManager.handleResponse(GetManager.java:224)
	at com.github.ambry.router.NonBlockingRouter$OperationController.onResponse(NonBlockingRouter.java:688)
	at com.github.ambry.router.NonBlockingRouter$OperationController.run(NonBlockingRouter.java:721)
	at java.lang.Thread.run(Thread.java:745)

```